Version 0.6.0 breaks on EL8

Tried to upgrade the plugin to 0.6.0 on some EL8 (AlmaLinux) systems : the plugin fails with this error
```
[root@ct-wn-1 ~]# /opt/nomad/plugins/nomad-driver-podman
/opt/nomad/plugins/nomad-driver-podman: /lib64/libc.so.6: version `GLIBC_2.33' not found (required by /opt/nomad/plugins/nomad-driver-podman)
/opt/nomad/plugins/nomad-driver-podman: /lib64/libc.so.6: version `GLIBC_2.32' not found (required by /opt/nomad/plugins/nomad-driver-podman)
/opt/nomad/plugins/nomad-driver-podman: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by /opt/nomad/plugins/nomad-driver-podman)
[root@ct-wn-1 ~]# 
```
On EL8, glibc version is 2.28

This prevents nomad from starting, even if the plugin is disabled (as Nomad still tries to fingerprint it) :

```
juil. 15 14:43:35 ct-wn-1 nomad[2342912]: ==> Loaded configuration from /opt/nomad/etc/nomad.hcl, /opt/nomad/etc/reload.hcl
juil. 15 14:43:35 ct-wn-1 nomad[2342912]: ==> Starting Nomad agent...
juil. 15 14:43:35 ct-wn-1 nomad[2342912]: ==> Error starting agent: failed to create plugin loader: failed to initialize plugin loader: failed to fingerprint plugins: 1 error occurred:
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:         * Unrecognized remote plugin message:
juil. 15 14:43:35 ct-wn-1 nomad[2342912]: Failed to read any lines from plugin's stdout
juil. 15 14:43:35 ct-wn-1 nomad[2342912]: This usually means
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   the plugin was not compiled for this architecture,
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   the plugin is missing dynamic-link libraries necessary to run,
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   the plugin is not executable by this process due to file permissions, or
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   the plugin failed to negotiate the initial go-plugin protocol handshake
juil. 15 14:43:35 ct-wn-1 nomad[2342912]: Additional notes about plugin:
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   Path: /opt/nomad/plugins/nomad-driver-podman
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   Mode: -rwxr-xr-x
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   Owner: 0 [root] (current: 0 [root])
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   Group: 0 [root] (current: 0 [root])
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   ELF architecture: EM_X86_64 (current architecture: amd64)
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:     2024-07-15T14:43:35.171+0200 [WARN]  agent.plugin_loader: plugin not referenced in the agent configuration file, future versions of Nomad will not load this plugin until the agent configuratio
n is updated: plugin_dir=/opt/nomad/plugins plugin=nomad-driver-podman
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:     2024-07-15T14:43:35.172+0200 [ERROR] agent.plugin_loader: plugin process exited: plugin_dir=/opt/nomad/plugins plugin=/opt/nomad/plugins/nomad-driver-podman id=2342931 error="exit status 1"
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:     2024-07-15T14:43:35.172+0200 [WARN]  agent.plugin_loader: plugin failed to exit gracefully: plugin_dir=/opt/nomad/plugins
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:     2024-07-15T14:43:35.172+0200 [ERROR] agent.plugin_loader: failed to fingerprint plugin: plugin_dir=/opt/nomad/plugins plugin=nomad-driver-podman
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   error=
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   | Unrecognized remote plugin message:
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   | Failed to read any lines from plugin's stdout
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   | This usually means
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   |   the plugin was not compiled for this architecture,
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   |   the plugin is missing dynamic-link libraries necessary to run,
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   |   the plugin is not executable by this process due to file permissions, or
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   |   the plugin failed to negotiate the initial go-plugin protocol handshake
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   |
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   | Additional notes about plugin:
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   |   Path: /opt/nomad/plugins/nomad-driver-podman
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   |   Mode: -rwxr-xr-x
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   |   Owner: 0 [root] (current: 0 [root])
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   |   Group: 0 [root] (current: 0 [root])
juil. 15 14:43:35 ct-wn-1 nomad[2342912]:   |   ELF architecture: EM_X86_64 (current architecture: amd64)

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,9 @@
 ## UNRELEASED
 
+BUG FIXES:
+
+* build: Removed CGO dependency accidentally introduced in 0.6.0
+
 ## 0.6.0 (July 12, 2024)
 
 IMPROVEMENTS:

--- a/GNUmakefile
+++ b/GNUmakefile
@@ -102,6 +102,7 @@ endif
 dist/%/nomad-driver-podman: GO_OUT ?= $@
 dist/%/nomad-driver-podman:
 	@echo "==> RELEASE BUILD of $@ ..."
+	CGO_ENABLED=0 \
 	GOOS=linux GOARCH=$(lastword $(subst _, ,$*)) \
 	go build -trimpath -o $(GO_OUT)
 

--- a/GNUmakefile
+++ b/GNUmakefile
@@ -8,7 +8,6 @@ BIN := $(shell go env GOBIN)
 ifndef BIN
 BIN := $(GOPATH)/bin
 endif
-
 default: help
 
 HELP_FORMAT="    \033[36m%-25s\033[0m %s\n"
@@ -54,26 +53,13 @@ clean: ## Cleanup previous build
 	@echo "==> Cleanup previous build"
 	rm -f ./build/nomad-driver-podman
 
-pkg/%/nomad-driver-podman: GO_OUT ?= $@
-pkg/%/nomad-driver-podman: ## Build the nomad-driver-podman plugin for GOOS_GOARCH, e.g. pkg/linux_amd64/nomad-driver-podman
-	@echo "==> Building $@ with tags $(GO_TAGS)..."
-	CGO_ENABLED=0 \
-		GOOS=$(firstword $(subst _, ,$*)) \
-		GOARCH=$(lastword $(subst _, ,$*)) \
-		go build -trimpath -o $(GO_OUT)
-
-.PRECIOUS: pkg/%/nomad-driver-podman
-pkg/%.zip: pkg/%/nomad-driver-podman ## Build and zip the nomad-driver-podman plugin for GOOS_GOARCH, e.g. pkg/linux_amd64.zip
-	@echo "==> Packaging for $@..."
-	@cp LICENSE $(dir $<)LICENSE.txt
-	zip -j $@ $(dir $<)*
-
 .PHONY: dev
 dev: clean build/nomad-driver-podman ## Build the nomad-driver-podman plugin
 
 build/nomad-driver-podman:
 	@echo "==> Building driver plugin ..."
 	mkdir -p build
+	CGO_ENABLED=0 \
 	go build -o build/nomad-driver-podman .
 
 .PHONY: test

